sdk/src/types/block/basic.rs

```diff
@@ -0,0 +1,167 @@
+    pub(crate) shallow_like_parents: ShallowLikeParents,
+    /// The optional [Payload] of the block.
+    pub(crate) payload: OptionalPayload,
+    /// The amount of mana the Account identified by [`IssuerId`](super::IssuerId) is at most
+    /// willing to burn for this block.
+    pub(crate) burned_mana: u64,
+}
+
+impl BasicBlock {
+    pub const KIND: u8 = 0;
+
+    /// Returns the strong parents of a [`BasicBlock`].
+    #[inline(always)]
+    pub fn strong_parents(&self) -> &StrongParents {
+        &self.strong_parents
+    }
+
+    /// Returns the weak parents of a [`BasicBlock`].
+    #[inline(always)]
+    pub fn weak_parents(&self) -> &WeakParents {
+        &self.weak_parents
+    }
+
+    /// Returns the shallow like parents of a [`BasicBlock`].
+    #[inline(always)]
+    pub fn shallow_like_parents(&self) -> &ShallowLikeParents {
+        &self.shallow_like_parents
+    }
+
+    /// Returns the optional payload of a [`BasicBlock`].
+    #[inline(always)]
+    pub fn payload(&self) -> Option<&Payload> {
+        self.payload.as_ref()
+    }
+
+    /// Returns the burned mana of a [`BasicBlock`].
+    #[inline(always)]
+    pub fn burned_mana(&self) -> u64 {
+        self.burned_mana
+    }
+}
+
+impl Packable for BasicBlock {
+    type UnpackError = Error;
+    type UnpackVisitor = ProtocolParameters;
+
+    fn pack<P: Packer>(&self, packer: &mut P) -> Result<(), P::Error> {
+        self.strong_parents.pack(packer)?;
+        self.weak_parents.pack(packer)?;
+        self.shallow_like_parents.pack(packer)?;
+        self.payload.pack(packer)?;
+        self.burned_mana.pack(packer)?;
+
+        Ok(())
+    }
+
+    fn unpack<U: Unpacker, const VERIFY: bool>(
+        unpacker: &mut U,
+        visitor: &Self::UnpackVisitor,
+    ) -> Result<Self, UnpackError<Self::UnpackError, U::Error>> {
+        let strong_parents = StrongParents::unpack::<_, VERIFY>(unpacker, &())?;
+        let weak_parents = WeakParents::unpack::<_, VERIFY>(unpacker, &())?;
+        let shallow_like_parents = ShallowLikeParents::unpack::<_, VERIFY>(unpacker, &())?;
+
+        if VERIFY {
+            verify_parents(&strong_parents, &weak_parents, &shallow_like_parents).map_err(UnpackError::Packable)?;
+        }
+
+        let payload = OptionalPayload::unpack::<_, VERIFY>(unpacker, visitor)?;
+
+        let burned_mana = u64::unpack::<_, VERIFY>(unpacker, &()).coerce()?;
+
+        let block = Self {
+            strong_parents,
+            weak_parents,
+            shallow_like_parents,
+            payload,
+            burned_mana,
+        };
+
+        Ok(block)
+    }
+}
+
+pub(crate) mod dto {
+    use alloc::collections::BTreeSet;
+
+    use serde::{Deserialize, Serialize};
+
+    use super::*;
+    use crate::types::{
+        block::{payload::dto::PayloadDto, BlockId, Error},
+        TryFromDto, ValidationParams,
+    };
+
+    /// The block object that nodes gossip around in the network.
```

Comment:
Same, maybe a more fitting comment in the TIP? This one should be fore the wrapper I guess